### PR TITLE
[ie/Hotstar] Downloading Thumbnails download Added

### PR DESCRIPTION
**Pull request by Rajeshwaran2001**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Added support to download thumbnails.

Fixes #

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by seproDev** [comment]
Thank you for your PR. Sadly, as it is, this cannot be merged. Don't use `requests` directly, but instead use helper functions like `self._download_json`. There are also numerous other style violations, so please take another look at the [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions).

**Comment by Rajeshwaran2001** [comment]
> Thank you for your PR. Sadly, as it is, this cannot be merged. Don't use `requests` directly, but instead, use helper functions like `self._download_json`. There are also numerous other style violations, so please take another look at the [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions).

I tried self._download_json and got 503. If you are available in Discord please share your username or share any other. Thanks in Advance.

**Comment by seproDev** [comment]
If you have questions regarding extractor development, you can ask in #yt-dlp-developers on the [yt-dlp discord server](https://discord.gg/H5MNcFW63r).

I'm not super familiar with the hotstar extractor, but given that there are call_api methods that look very similar to what you are doing, those should be used/expanded. I'm also not sure if making two additional requests for thumbnails is worth it. Is there no other way to retrieve them?

**Comment by Rajeshwaran2001** [comment]
> If you have questions regarding extractor development, you can ask in #yt-dlp-developers on the [yt-dlp discord server](https://discord.gg/H5MNcFW63r).
> 
> I'm not super familiar with the Hotstar extractor, but given that there are call_api methods that look very similar to what you are doing, those should be used/expanded. I'm also not sure if making two additional requests for thumbnails is worth it. Is there no other way to retrieve them?

video_data has image sets but does not have the complete URL and I want to know which call_api methods to perform POST request

**Comment by seproDev** [comment]
> video_data has image sets but does not have the complete URL

Yeah, looking at the API response of the `/detail` call, the thumbnail path is already included. So just use that.

**Comment by seproDev** [comment]
Those changes don't make sense. `imageSets` looks like this:
```js
'imageSets': {
  'DEFAULT': {
    'h': 'sources/r1/cms/prod/1933/1631933-h-1eeb2ec30740'
  }
}
```
So you want something like this:
```python
'thumbnail': traverse_obj(video_data, 'imageSets', 'DEFAULT', 'h',
                          {lambda x: x and f'https://img10.hotstar.com/image/upload/f_auto/{x}'}),
```
Please also revert the changes to new lines and remove the unused import.

File: yt_dlp/extractor/hotstar.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import re
 import time
 import uuid
+from urllib.parse import urlparse
```
Comment (pukkandan) [inline yt_dlp/extractor/hotstar.py:7]:
```suggestion
```

File: yt_dlp/extractor/hotstar.py [hunk 2]
```diff
@@ -242,6 +244,7 @@ def _real_extract(self, url):
 
         # See https://github.com/yt-dlp/yt-dlp/issues/396
         st = self._download_webpage_handle(f'{self._BASE_URL}/in', video_id)[1].headers.get('x-origin-date')
+
```
Comment (pukkandan) [inline yt_dlp/extractor/hotstar.py:247]:
```suggestion
```

File: yt_dlp/extractor/hotstar.py [hunk 3]
```diff
@@ -232,7 +234,6 @@ def _real_extract(self, url):
         video_id, video_type = self._match_valid_url(url).group('id', 'type')
         video_type = self._TYPE.get(video_type, video_type)
         cookies = self._get_cookies(url)  # Cookies before any request
-
```
Comment (pukkandan) [inline yt_dlp/extractor/hotstar.py:235]:
```

```

File: yt_dlp/extractor/hotstar.py [hunk 4]
```diff
@@ -332,6 +333,7 @@ def _real_extract(self, url):
             'season_id': video_data.get('seasonId'),
             'episode': video_data.get('title'),
             'episode_number': int_or_none(video_data.get('episodeNo')),
+            'thumbnails': video_data.get('imageSets'),
```
Comment (pukkandan) [inline yt_dlp/extractor/hotstar.py:336]: Transform the data to correct field names we want using `traverse_obj`. See docstring of InfoExtractor in common.py

File: yt_dlp/extractor/hotstar.py (hunk 3 above)
Comment (Rajeshwaran2001) [reply yt_dlp/extractor/hotstar.py:235]: Sorry that was a wrong commit

**Comment by Rajeshwaran2001** [comment]
@bashonly  Is the new update fine to merge?

**Pull request closed without merging**